SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  NotificationFactory.swift
+//  SCSH_3C_iOS
+//
+//  Created by 辜敬閎 on 2023/9/17.
+//
+
+import UserNotifications
+
+protocol NotificationFactoryType {
+    func generate(identifier: String, title: String, body: String, triggerInterval: TimeInterval) -> UNNotificationRequest
+}
+
+extension NotificationFactoryType {
+    func generate(identifier: String = UUID().uuidString, title: String, body: String, triggerInterval: TimeInterval = 1) -> UNNotificationRequest {
+        generate(identifier: identifier, title: title, body: body, triggerInterval: triggerInterval)
```

Comment:
Is this an idiom of Swift extensions, calling a function with the same name within a function?

Reply:
In Swift, this is how we implement `Overloading`.
In reality, the function signatures are distinct, as you can see, one of them includes a default value of `UUID().uuidString`.

Reply:
Regarding whether it's an idiom or not?

I'd say... maybe? after all, the Swift compiler DOES warn you if you cause any infinite recursion.

I think that is your concern?

Reply:
Oh.... It makes sense if this is overloading. I am OK with it.